File: python/runner.py
#!/usr/bin/env python
import sys
import ujson
import time

# handle command line arguments
if len(sys.argv) != 3:
    print "Usage: ./test_python <input> <output>"
    sys.exit(1)

# read and parse input
read_start = time.time()
with open(sys.argv[1], "r") as f1:
    text = f1.read()
    parse_start = time.time()
    data = ujson.loads(text)
    parse_end = time.time()
read_end = time.time()


# serialize and write output
write_start = time.time()
with open(sys.argv[2], "w") as f2:
    serialize_start = time.time()
    serialized_data = ujson.dumps(data)
    serialize_end = time.time()
    f2.write(serialized_data)
write_end = time.time()

# stats
print "r:" + str(read_end - read_start)
print "p:" + str(parse_end - parse_start)
print "w:" + str(write_end - write_start)
print "s:" + str(serialize_end - serialize_start)

File: cpp/runner.c
#include "rapidjson/document.h"
#include "rapidjson/writer.h"
#include "rapidjson/stringbuffer.h"
#include <iostream>
#include "rapidjson/filereadstream.h"
#include "rapidjson/filewritestream.h"
#include <cstdio>
#include <ctime>

using namespace rapidjson;

int main(int argc, char** argv) {
    // handle command line arguments
    if(argc != 3) {
        std::cout << "Usage: ./test_cpp <input> <output>" << std::endl;
        return 1;
    }

    // read and parse input
	std::clock_t read_start = std::clock();
    char readBuffer[65536];
    FILE* input_file = fopen(argv[1], "rb");
    FileReadStream is(input_file, readBuffer, sizeof(readBuffer));
    Document d;
    std::clock_t parse_start = std::clock();
    d.ParseStream(is);
    std::clock_t parse_end = std::clock();
    fclose(input_file);
    std::clock_t read_end = std::clock();

    // serialize and write output
    std::clock_t write_start = std::clock();
    FILE* output_file = fopen(argv[2], "wb");
	char writeBuffer[65536];
	FileWriteStream os(output_file, writeBuffer, sizeof(writeBuffer));
    std::clock_t serialize_start = std::clock();
	Writer<FileWriteStream> writer(os);
	d.Accept(writer);
    std::clock_t serialize_end = std::clock();
	fclose(output_file);
	std::clock_t write_end = std::clock();

    // stats
	std::cout << "r:" << (read_end - read_start) / (double) (CLOCKS_PER_SEC) << std::endl;
    std::cout << "p:" << (parse_end - parse_start) / (double) (CLOCKS_PER_SEC) << std::endl;
	std::cout << "w:" << (write_end - write_start) / (double) (CLOCKS_PER_SEC ) << std::endl;
    std::cout << "s:" << (serialize_end - serialize_start) / (double) (CLOCKS_PER_SEC) << std::endl;
    return 0;
}